TestResultSummaryService/DataManager.js

```diff
@@ -49,6 +50,44 @@ class DataManager {
         return -1;
     }
 
+    async updateApplicationTests( data ) {
+        const { testName, testData, buildName, timestamp, _id, ...newData } = data;
+        if ( testData && testData.externalTestExtraData ) {
+            const buildInfo = Utils.getInfoFromBuildName(buildName);
+            if ( buildInfo ) {
+                const appDB = new ApplicationTestsDB();
+                const { dockerOS, appVersion, appName, javaVersionOutput } = testData.externalTestExtraData;
+                const { jdkVersion, jdkImpl, platform } = buildInfo;
+                let query = {
+                    testName,
+                    jdkVersion,
+                    jdkImpl,
+                    jdkPlatform: platform,
+                    dockerOS,
+                    appVersion,
+                    appName,
+                };
+                const update = {
+                    testId: _id,
+                    timestamp,
+                    javaVersionOutput,
+                    buildName,
+                    ...newData
+                }
+                const existingData = await appDB.getData(query).toArray();
+                if ( existingData && existingData.length > 0 ) {

```

Comment:
I would expect `existingData.length === 1`. If not, then we will have a problem.

Reply:
updated